lib/galaxy/datatypes/interval.py

```diff
@@ -1482,6 +1482,86 @@ def sniff( self, filename ):
+    """
+    Support for the Strand-specific coordinate count datatype which is used
+    by tools within the Chip-exo Galaxy flavor.  The Chip-exo Galaxy flavor
+    is used by the Center for Eukaryotic Gene Regulation labs at Penn State
+    University.
+
+    ScIdx files are 1-based and consist of strand-specific coordinate counts.
+    They always have 5 columns, and the first row is the column labels:
+    'chrom', 'index', 'forward', 'reverse', 'value'.
+    Each line following the first consists of data:
+    chromosome name (type str), peak index (type int), Forward strand peak
+    count (type int), Reverse strand peak count (type int) and value (type int).
+    The value of the 5th 'value' column is the sum of the forward and end
+    reverse peak count values.
+    """
+    file_ext = "scidx"
+
+    MetadataElement(name="columns", default=0, desc="Number of columns", readonly=True, visible=False)
+    MetadataElement(name="column_types", default=[], param=metadata.ColumnTypesParameter, desc="Column types", readonly=True, visible=False, no_value=[])
+
+    def __init__(self, **kwd):
+        """
+        Initialize scidx datatype.
+        """
+        Tabular.__init__(self, **kwd)
+        # Don't set column names since the first
+        # line of the dataset displays them.
+        self.column_names = []
+
+    def sniff(self, filename):
+        """
+        Checks for 'scidx-ness.'
+        """
+
+        def is_scidx(fh, bool):
+            fh.close()
+            return bool
+
+        try:
+            count = 0
+            fh = open(filename, "r")
+            while True:
+                line = fh.readline()
+                line = line.strip()
+                if not line:
+                    # EOF
+                    if count > 1:
+                        # We need at least the column labels and a data line.
+                        return is_scidx(fh, True)
+                    return is_scidx(fh, False)
+                # Skip first line.
+                if count > 0:
+                    items = line.split('\t')
+                    if len(items) !=5:
+                        return is_scidx(fh, False)
+                    index = items[1]
+                    if not index.isdigit():
+                        return is_scidx(fh, False)
+                    forward = items[2]
+                    if not forward.isdigit():
+                        return is_scidx(fh, False)
+                    reverse = items[3]
+                    if not reverse.isdigit():
+                        return is_scidx(fh, False)
+                    value = items[4]
+                    if not value.isdigit():
+                        return is_scidx(fh, False)
+                    if int(forward) + int(reverse) != int(value):
+                        return is_scidx(fh, False)
+                if count == 100:
+                    return is_scidx(fh, True)
+                count += 1
+            if count < 100 and count > 0:
```

Comment:
Actually, this case covers the scenario where the files is less than 100 lines, but has not yet failed any checks, so this check is outside of the while.  The scenario where the files is longer than 100 lines, but has not failed any checks is handled by the "if count == 100:" check 3 lines above.